timemachine/potentials/jax_utils.py

```diff
@@ -3,10 +3,8 @@
 from jax import vmap
 from numpy.typing import NDArray
 
-Array = NDArray
 
-
-def get_all_pairs_indices(n: int) -> Array:
+def get_all_pairs_indices(n: int) -> NDArray:
```

Comment:
Question: Does mypy not allow type aliases like this? (Or perhaps there's a preferred way to declare type aliases?)

Reply:
I was surprised by this too -- there doesn't seem to be a problem with type aliases in general, but a naive `Array = NDArray` does not work; this results in `Variable "timemachine.potentials.jax_utils.Array" is not valid as a type` wherever the alias is used. Still trying to understand the reason for this and whether there's a solution that still allows aliasing..

Reply:
The issue seems to be that mypy doesn't correctly identify this as a type alias (relevant [mypy issue](https://github.com/python/mypy/issues/7866)).

[PEP 613](https://peps.python.org/pep-0613/) seems to fix this by allowing explicit specification of type aliases (via the `TypeAlias` type), and an implementation is available for 3.7 in the `typing-extensions` backport package. I added `typing-extensions` and reverted the removal of the `Array` alias in 37d5f5e.

Reply:
Nice! Thanks for looking into that -- the  `MyT = T` -> `MyT: TypeAlias = T` fix looks easy (and imo improves clarity)